### PR TITLE
429203: add coreai-jaf-frontend service files

### PR DESCRIPTION
**Pull request by J0hn-B-es**
# **What this PR does / why we need it**:
Add the files for the coreai-jaf-frontend service 

*Link to the relevant work items: *
[AB#429203](https://dev.azure.com/defragovuk/DEFRA-FFC/_workitems/edit/429203)

# **Special notes for your reviewer**
*Any specific actions or notes on review?*

# Testing
*Any relevant testing information and pipeline runs.*
image

# **How does this PR make you feel**:
gif

### CODE REVIEW

**Review by rajeshkaremane — COMMENTED** [review]
1. Please link the ticket : Refer the old PR for the PR template: https://github.com/DEFRA/adp-infrastructure-services/pull/54
2. Run the pipeline in SND environment and ensure the change works
3. Optionally you can use the common.yaml instead of having each environment specific files

File: infra/services/coreai/coreai-jaf-frontend/env-vars/snd4.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+variables:
+  appEndpoint: "coreai-jaf-frontend"
```
Comment (rajeshkaremane) [inline infra/services/coreai/coreai-jaf-frontend/env-vars/snd4.yaml:2]: https://coreai-jaf-frontend.snd4.adp.defra.gov.uk/ 
This link is not working - need to check if the domain should be 
https://crai-jaf-frontend.snd4.adp.defra.gov.uk/

Comment (J0hn-B-es) [reply infra/services/coreai/coreai-jaf-frontend/env-vars/snd4.yaml]: The domain has been updated to https://crai-jaf-frontend.snd4.adp.defra.gov.uk/

File: infra/services/coreai/deploy-services-infra.yaml [hunk 2]
```diff
@@ -29,7 +29,7 @@ parameters:
       - coreai-mcu-frontend
       - coreai-mcu-generation-oai
       - coreai-mcu-knowledge-pgv
-      - coreai-jaf-frontend
+      - crai-jaf-frontend
```
Comment (DaRoszko) [inline infra/services/coreai/deploy-services-infra.yaml:32]: this is OK, just noy strictly needed in this file

**Comment by rajeshkaremane** [comment]
1. Update the branch on the PR.
2. Is the internal vs default issue fixed? Unless that is fixed first, I don't think this PR can be merged

**Review by rajeshkaremane — COMMENTED** [review]
image

Above issue need to be fixed before the PR gets merged

**Comment by njmoorewitchcat** [comment]
Indeed - we are going to complete ticket  AB#434381 before merging this.  If we do not, then the service will deploy to DEV1 and the WAF rule will fail.  This PR will not be fully approved and merged until then.

https://dev.azure.com/defragovuk/DEFRA-FFC/_workitems/edit/

**Review by farshid3003 — APPROVED** [review]

**Review by farshid3003 — APPROVED** [review]

File: infra/services/coreai/deploy-services-infra.yaml [hunk 3]
```diff
@@ -55,7 +56,7 @@ extends:
     service: ${{ parameters.service }}
     ingressType: ${{ parameters.ingressType }}
     resource: ${{ parameters.resource }}
-    ${{ if or(eq(parameters.service, 'coreai-mcu-generation-oai'),eq(parameters.service, 'coreai-mcu-knowledge-pgv') ) }}:
-      includeServiceVariableFiles: Common
+    ? ${{ if or(eq(parameters.service, 'coreai-mcu-generation-oai'),eq(parameters.service, 'coreai-mcu-knowledge-pgv') ) }}
```
[the remark this answers is not in the document]
Comment (J0hn-B-es) [reply infra/services/coreai/deploy-services-infra.yaml:59]: yes, please check [here](https://yaml.org/spec/1.2.2/#53-indicator-characters)

**Review by kstockdalEE — APPROVED** [review]

**Pull request merged**